soh/include/z64item.h

```diff
@@ -244,6 +244,7 @@ typedef enum {
     /* 0x99 */ ITEM_STICK_UPGRADE_30,
     /* 0x9A */ ITEM_NUT_UPGRADE_30,
     /* 0x9B */ ITEM_NUT_UPGRADE_40,
+    /* 0x9C */ ITEM_TRIFORCE_PIECE,
```

Comment:
I could be wrong, but I don't think we're supposed to be adding anything to these enums, this is a randomizer specific item cc @leggettc18

Reply:
Yes, and in the case of randomizer-specific items, the RandomizerCheck is used as the ItemID and GI_ID in the Randomizer item table (and possibly the GID? not so sure about that one), so the GI_TRIFORCE_PIECE down further is also not supposed to be added, if I remember the item tables correctly.

Reply:
I needed `ITEM_TRIFORCE_PIECE` specifically to add it to the item tracker, which uses these enums. `GI_TRIFORCE_PIECE` is used on the 3DS rando side. For both accounts, triforce pieces are the first truly new item in rando, so we don't really have a precedence or pattern set up for this.

Reply:
I could be misunderstanding the requirements, but it looks like on https://github.com/HarbourMasters/Shipwright/pull/3082 they've added entirely new items without touching the vanilla enums

Reply:
> I could be misunderstanding the requirements, but it looks like on #3082 they've added entirely new items without touching the vanilla enums

They haven't added an item tracker entry, which is what's using the ITEM_ enums, or added the new items as ice traps, which is what's using the GI_ enums. I don't know how we would gracefully handle this, but using the already existing GID enum or other enums that are randomizer specific could clash with other vanilla enums if they happen to be the same as the randomizer specific enums.

Reply:
I think the item tracker can easily be updated to allow for randomizer items by just updating this struct in `randomizer_item_tracker.h` (and the corresponding define) to include a modIndex:
```cpp
typedef struct ItemTrackerItem {
    uint32_t id;
    std::string name;
    std::string nameFaded;
    uint32_t data;
    void (*drawFunc)(ItemTrackerItem);
} ItemTrackerItem;
```
I was planning on changing it in a followup pr to https://github.com/HarbourMasters/Shipwright/pull/3082.

Reply:
Really all that needs to be done is providing a custom drawFunc and to load the required texture resource into ImGui (currently happening in debugSaveEditor.cpp).

You don't have to have an ITEM_ enum, you just need something unique enough for the possible inputs to a new DrawRandomizerItem type function (like RG_ enum probably).

Reply:
> Really all that needs to be done is providing a custom drawFunc and to load the required texture resource into ImGui (currently happening in debugSaveEditor.cpp).
> 
> You don't have to have an ITEM_ enum, you just need something unique enough for the possible inputs to a new DrawRandomizerItem type function (like RG_ enum probably).

I guess I could use the RG_ enum in this case as it's already higher than the highest ITEM_ enum, but is that something we want to be relying on?

Reply:
Something similar is already used for ice traps, the models are based on the gi enum but values above the vanilla max (0x7D) are used for randomizer items:
```cpp
IceTrapModels = {
    GI_SHIELD_MIRROR,
    GI_BOOMERANG,
    GI_LENS,
    GI_HAMMER,
    GI_BOOTS_IRON,
    GI_BOOTS_HOVER,
    GI_STONE_OF_AGONY,
    GI_DINS_FIRE,
    GI_FARORES_WIND,
    GI_NAYRUS_LOVE,
    GI_ARROW_FIRE,
    GI_ARROW_ICE,
    GI_ARROW_LIGHT,
    0xB8, //Double defense
    GI_CLAIM_CHECK,
    0x80, //Progressive hookshot
    0x81, //Progressive strength
    0x82, //Progressive bomb bag
    0x83, //Progressive bow
    0x84, //Progressive slingshot
    0x85, //Progressive wallet
    0x86, //Progressive scale
    0x8A, //Progressive magic
  };
```

Reply:
Welp guess I completely missed that. I might be able to just use the GID for the ice traps as well then, I'll take a look.

Reply:
Last commit should remove both ITEM_ and GI_ usages. I'm *really* not a big fan of how randomizer is currently assigning randomizer items to the ice trap models though, but I guess for now I'm following what was already being used for that. This should definitely be cleaned up at some point though, this not using enums whatsoever makes it so easy to accidentily use a number that's already in use by something else.